src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c

```diff
@@ -705,6 +734,43 @@ cJSON *wm_vuldet_decode_advisories(char *advisories) {
     return json_advisories;
 }
 
+int wm_vuldet_add_cve_node(cve_vuln_pkg *newPkg, const char *cve, OSHash *cve_table) {
+    int status = 0;
+    cve_vuln_pkg *last_pkg = NULL;
+
+    if (status = OSHash_Add(cve_table, cve, (void *)newPkg), !status) {
+        return -1;
+    }
+
+    // If the cve node already exists, add the pkg to the end of the linked list.
+    if (status == 1) {
+        if (last_pkg = (cve_vuln_pkg *)OSHash_Get(cve_table, cve), !last_pkg) {
+            return -1;
+        }
+
+        while (last_pkg->next) last_pkg = last_pkg->next;
```

Comment:
You should check if `last_package` and `newPkg` have the same `src_name`. If that is the case, check if they have the same `bin_name`. If it is the same, mark that `bin_name` as duplicated. If not, add the `bin_name` to the list.

Reply:
Agree! The function should handle duplicate packages within the linked list.